collect_app/src/test/java/org/odk/collect/android/preferences/AggregatePreferencesAdderTest.java

```diff
@@ -3,24 +3,27 @@
 import android.preference.PreferenceFragment;
 import android.preference.PreferenceManager;
 
+import androidx.preference.PreferenceFragmentCompat;
+import androidx.test.core.app.ApplicationProvider;
+
 import org.junit.Test;
 import org.junit.runner.RunWith;
 import org.odk.collect.android.R;
 import org.robolectric.Robolectric;
 import org.robolectric.RobolectricTestRunner;
-import org.robolectric.RuntimeEnvironment;
 import org.robolectric.shadows.ShadowToast;
 
 import static junit.framework.Assert.assertFalse;
 import static org.junit.Assert.assertEquals;
 import static org.junit.Assert.assertTrue;
+import static org.mockito.Mockito.mock;
 
 @RunWith(RobolectricTestRunner.class)
 public class AggregatePreferencesAdderTest {
 
     @Test
     public void whenPreferencesAreAdded_returnsTrue() {
-        PreferenceFragment fragment = Robolectric.buildFragment(TestPreferenceFragment.class).create().resume().visible().get();
+        PreferenceFragmentCompat fragment = mock(PreferenceFragmentCompat.class);
         AggregatePreferencesAdder loader = new AggregatePreferencesAdder(fragment);
```

Comment:
Hi @seadowg! I was not able to create `PreferenceFragmentCompat` using Robolectric, and could not think of any other method to test it. I made `BasePreferenceFragment` extract `PreferenceFragmentCompat`, as `androidx.PreferenceFragment` is also deprecated.

Reply:
Mocking is fine here I think! You can make the call to `addPreferencesFromResource` throw a `ClassCastException` using Mockito's `thenThrow`.

Reply:
Oh okay Thanks @seadowg!